Exception raised on module version conflicts

In case of a conflict in module versions, a `GroovyRuntimeException` is raised while a simple warning was expected:

```
Caused by: groovy.lang.GroovyRuntimeException: Conflicting module versions. Module [groovy-xml is loaded in version 4.0.6 and you are trying to load version 4.0.13
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl$DefaultModuleListener.onModule(MetaClassRegistryImpl.java:524)
	at org.codehaus.groovy.runtime.m12n.ExtensionModuleScanner.scanExtensionModuleFromProperties(ExtensionModuleScanner.java:87)
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.registerExtensionModuleFromProperties(MetaClassRegistryImpl.java:176)
	at io.quarkiverse.groovy.runtime.GroovyRecorder.registerExtensionModuleFromProperties(GroovyRecorder.java:78)
	at io.quarkiverse.groovy.runtime.GroovyRecorder.scanExtensionModuleFromMetaInf(GroovyRecorder.java:69)
	at io.quarkiverse.groovy.runtime.GroovyRecorder.scanModulesFrom(GroovyRecorder.java:58)
	at io.quarkiverse.groovy.runtime.GroovyRecorder.initExtensionModules(GroovyRecorder.java:48)
	at io.quarkus.deployment.steps.GroovyProcessor$initExtensionModules1008776114.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.GroovyProcessor$initExtensionModules1008776114.deploy(Unknown Source)
```

--- a/.github/project.yml
+++ b/.github/project.yml
@@ -1,4 +1,4 @@
 release:
-  current-version: 3.2.0
+  current-version: 3.2.1
   next-version: "999-SNAPSHOT"
 

--- a/extensions/core/runtime/src/main/java/io/quarkiverse/groovy/runtime/GroovyRecorder.java
+++ b/extensions/core/runtime/src/main/java/io/quarkiverse/groovy/runtime/GroovyRecorder.java
@@ -16,7 +16,6 @@
  */
 package io.quarkiverse.groovy.runtime;
 
-import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
 import java.util.Enumeration;
@@ -57,7 +56,7 @@ public class GroovyRecorder {
                 URL url = resources.nextElement();
                 scanExtensionModuleFromMetaInf(url);
             }
-        } catch (IOException e) {
+        } catch (Exception e) {
             LOG.warnf("An error occurred while scanning the extension modules '%s': %s", moduleMetaInfFile, e.getMessage());
         }
     }
@@ -67,7 +66,7 @@ public class GroovyRecorder {
             Properties properties = new Properties();
             properties.load(inStream);
             registerExtensionModuleFromProperties(properties);
-        } catch (IOException e) {
+        } catch (Exception e) {
             LOG.warnf("An error occurred while registering the extension module '%s': %s", metadata, e.getMessage());
         }
     }

